src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figuresArray = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < figuresArray.length; i++) {
+            if (i > Math.floor(figuresArray.length / 2)) {
+                figuresArray[i] = figureSupplier.getRandomFigure(100);
+            } else {
+                figuresArray[i] = figureSupplier.getDefaultFigure();
+            }
```

Comment:
Creating a new 'FigureSupplier' instance inside the loop is inefficient if it is done in the 'getRandomFigure' method. Ensure that 'getRandomFigure' does not create a new 'FigureSupplier' on each call.